### ISSUE TITLE
#### Hello

### ISSUE THREAD

**Issue by stef30360** [quantopian/qgrid#156]
### Environment

* Operating System:
* Python Version: `$ python --version`
* How did you install Qgrid: (`pip`, `conda`, or `other (please explain)`)
* Python packages: `$ pip freeze` or `$ conda list` (please include qgrid, notebook, and jupyterlab versions)
* Jupyter lab packages (if applicable): `$ jupyter labextension list`

### Description of Issue

* What did you expect to happen?
* What happened instead?

### Reproduction Steps

1.
2.
3.
...

### What steps have you taken to resolve this already?

...

### Anything else?

...

**Comment by stef30360** [comment]
### It's possible to use ipywidget slider interactive with Qgridwidgets ?
thks

**Comment by TimShawver** [comment]
I had not tried doing that yet but I just got an example working where the slider widget acts as the filter for an integer column.

Given the following qgrid instance:
```
df_scale = pd.DataFrame(np.random.randn(1000, 4), columns=list('ABCD'))
q_scale = qgrid.show_grid(df_scale, show_toolbar=True)
q_scale
```

You can create a slider widget which controls the filter on the index column like this:
```
int_range = widgets.IntRangeSlider(
    value=[0, 999],
    min=0,
    max=999,
    step=1,
    description='Test:',
    disabled=False,
    continuous_update=True,
    orientation='horizontal',
    readout=True,
    readout_format='d',
)

def on_value_change(change):
    q_scale._handle_qgrid_msg_helper({
        'field': "index",
        'filter_info': {
            'field': "index",
            'max': change['new'][1],
            'min': change['new'][0],
            'type': "slider"
        },
        'type': "filter_changed"
    })

int_range.observe(on_value_change, names='value')
int_range
```
Now when you adjust the range of the slider, the grid's filter should update instantaneously. You should be able to do basically the same thing with FloatSlider for float columns.

Please note that the `_handle_qgrid_msg_helper` is not a documented part of the API so I would consider this functionality experimental at this point.

Does this answer question? I wasn't 100% sure what you wanted to do with the slider but this is the only reasonable thing I could think of doing with it so hopefully that's what you had in mind.

**Comment by outlace** [comment]
@TimShawver How can I filter a discrete integer column in the grid using your above method? I tried tweaking but it didn't work. I have a ipywidget BoundedInt widget and I want to filter a non-unique integer column (i.e. multiple rows may have the same value for this integer column).

**Comment by TimShawver** [comment]
I had trouble finding the `BoundedInt` widget in ipywidgets, seemed like it was a base class for `IntSlider`? Do you have a screen shot of the control or a link to docs I can look at for reference?

If it's similar to `IntSlider`, you'd presumably want to filter by just a single integer instead of a range, which would mean sending the same value for min/max, like this:
```
q_scale._handle_qgrid_msg_helper({
        'field': "field_name",
        'filter_info': {
            'field': "field_name",
            'max': change['new'],
            'min': change['new'],
            'type': "slider"
        },
        'type': "filter_changed"
    })
```

Also I wanted to mention that`any` is not a supported value for `type` (under `filter_info`). The possible values for that field are `slider`, `text`, `boolean`, and `date`.  Here is sample usage for the `filter_info` formats other than slider, which I already described above:

```
#Boolean:
qgrid_widget._handle_qgrid_msg_helper({
    'field': "field_name",
    'filter_info': {
        'field': "field_name",
        'selected': False,
        'type': 'boolean'
    },
    'type': "filter_changed"
})

#Text:
qgrid_widget._handle_qgrid_msg_helper({
    'field': "field_name",
    'filter_info': {
        'field': "field_name"
        'selected': [1, 3], # can also be set to 'all' in which case 'excluded' can be used
        'excluded': [] ,
        'type': 'text'
    },
    'type': "filter_changed"
})

#Date:
qgrid_widget._handle_qgrid_msg_helper({
    'field': "field_name",
    'filter_info': {
        'field': "field_name"
        'min': 1357775999999, # unit is milliseconds since the epoch
        'max': 1357171200000,
        'type': 'date'
    },
    'type': "filter_changed"
})

**Comment by outlace** [comment]
@TimShawver Thank you. I got it working.

Do you know how I can make an ipywidget button reset the filter for a column so it will show everything? I could use the min,max thing from above, but in general I won't know what the 'max' is.

**Comment by TimShawver** [comment]
Yep if you pass in `None` for `'min'` and `'max'`, that will reset the filter.  For example:
```
qgrid_widget._handle_qgrid_msg_helper({
    'field': "index",
    'filter_info': {
        'field': "index",
        'max': None,
        'min': None,
        'type': "slider"
    },
    'type': "filter_changed"
})
```

**Comment by luislourenzi** [comment]
Hi guys, 

I've tried to do this to modify in my grid an integer column named "Quantity", but it's not working. Can you help to check if I'm missing something? Thank you!

int_range = widgets.IntRangeSlider(
    value=[0, 200000],
    min=0,
    max=200000,
    step=1,
    description='Quantity:',
    disabled=False,
    continuous_update=True,
    orientation='horizontal',
    readout=True,
    readout_format='d',
)

def on_value_change(change):
    qgrid_widget._handle_qgrid_msg_helper({
        'field': "Quantity",
        'filter_info': {
            'field': "Quantity",
            'max': change['new'][1],
            'min': change['new'][0],
            'type': "slider"
        },
        'type': "filter_changed"
    })
    
int_range.observe(on_value_change, names='value')
int_range

**Comment by jfbeaumont** [comment]
Hey guys, it's now 2020 and I have a question around this.

I used the example:

![image](https://user-images.githubusercontent.com/692031/81112217-4f2d0d00-8eec-11ea-8cab-f62165578d32.png)

I added the following code

```
button = widgets.Button(description="filter!")

def on_button_clicked(b):
    #Text:
    qgrid_widget._handle_qgrid_msg_helper({
    'field': "D",
    'filter_info': {
        'field': "D",
        'selected': [0], # can also be set to 'all' in which case 'excluded' can be used
        'excluded': [] ,
        'type': 'text'
    },
    'type': "change_filter"})

button.on_click(on_button_clicked)

button
```

Got the following cell:

![image](https://user-images.githubusercontent.com/692031/81112400-987d5c80-8eec-11ea-8255-37566778a167.png)

But something weird happens. If I click on my button before filtering in the grid itself on column D, the swirl shows and the UI looks like it's stuck. However, if I filter first in the grid and then click my button, it works as expected:

![image](https://user-images.githubusercontent.com/692031/81112569-d8444400-8eec-11ea-8b1f-85c6d9681929.png)


Any clue about this? Many thanks!
